src/Castle.Windsor/MicroKernel/SubSystems/Conversion/TypeNameConverter.cs

```diff
@@ -144,6 +149,29 @@ private bool InitializeAppDomainAssemblies(bool forceLoad)
 					Scan(assembly);
 				}
 			}
+#else
+			if (assemblies.Count == 0)
+			{
+				var context = DependencyContext.Default;
```

Comment:
> The case where there were hundreds of assemblies, is this quite common on the projects you work on?

Nope not common, however I've heard of many other projects with large bin directories, however most tend to be thick clients so not really a problem for .NET Core with no WinForms since people can't migrate those to .NET Core.

> The worry here is we need to change this anyway, should we be concerned about how this(dep model stuff) executes under load?

They are obviously both going to perform poorly compared to the .NET Framework implementation if they eager load all assemblies.  However at least the `DependencyModel` stuff knows which assemblies to load and that info is already loaded by the `dotnet` CLI runtime host.

The other option since .NET Standard 2.0 is bringing `AppDomain.CurrentDomain.GetAssemblies()` back is to just use reflection to call it since coreclr has always had the APIs by the looks of it (just not exposed by corefx), we've done this many times in the past with different .NET Framework versions, it sucks but it'll mean we can more easily transition to .NET Standard 2.0 and avoid the other problems.

I'm just throwing my thoughts out there, if you feel strongly towards one approach let's talk on it.